finality-aleph/src/block/mod.rs

```diff
@@ -226,3 +235,24 @@ where
     /// Children of the specified block.
     fn children(&self, id: BlockId) -> Result<Vec<J::Header>, Self::Error>;
 }
+
+pub trait HeaderBackend<H: Header>: Send + Sync {
+    type Error: Debug;
+    /// Get block header. Returns `None` if block is not found.
+    fn header(&self, id: &BlockId) -> Result<Option<H>, Self::Error>;
+    /// Get hash of a finalized block with a given number. Returns Ok(None) if block exists in
+    /// the database, but is not finalized yet.
+    fn header_of_finalized_at(&self, number: BlockNumber) -> Result<Option<H>, Self::Error>;
+    /// Get currently highest finalized block.
+    fn top_finalized_id(&self) -> BlockId;
+    /// Get full id of a block with a given hash
+    fn hash_to_id(&self, hash: BlockHash) -> Result<Option<BlockId>, Self::Error>;
+}
+
+/// A strategy for selecting a leaf that should be considered as the chain tip.
```

Comment:
Calling it "the chain tip" introduces another name for ~the same concept. Either explain what this is about in detail ("for purposes of deciding which block to propose for finalization" or something like that) or use one of the names we already have ("best block" or "favourite block").

Reply:
Kinda agree, I was looking for more general term (`best` is somewhat associated with substrate's longest chain, and `favourite` with aleph's), but maybe let's not add 3rd name.